lib/ingram_micro/elements/return_authorization_submission.rb

```diff
@@ -1,67 +1,69 @@
-    @element[:customer] ||= IngramMicro::Customer.new
-    @element[:shipment_information] ||= IngramMicro::OutboundShipmentInformation.new
-    @element[:credit_card_information] ||= IngramMicro::CreditCardInformation.new
-    @element[:order_header] ||= IngramMicro::ReturnAuthorizationOrderHeader.new
-    @element[:purchase_order_information] ||= IngramMicro::OutboundPurchaseOrderInformation.new
-    @element[:detail] ||= IngramMicro::Detail.new
-  end
+    def initialize(options = {})
+      super
+      @element[:customer] ||= Customer.new
+      @element[:shipment_information] ||= OutboundShipmentInformation.new
+      @element[:credit_card_information] ||= CreditCardInformation.new
+      @element[:order_header] ||= ReturnAuthorizationOrderHeader.new
+      @element[:purchase_order_information] ||=
+        OutboundPurchaseOrderInformation.new
+      @element[:detail] ||= Detail.new
+    end
 
-  def build(builder)
-    builder.send('header') do
-      builder.send 'customer-id', @element[:customer_id]
-      builder.send 'business-name', @element[:business_name]
-      builder.send 'carrier-name', @element[:carrier_name]
-      builder.send('customer-information') do
-        @element[:customer].build(builder)
-      end
-      builder.send('shipment-information') do
-        @element[:shipment_information].build(builder)
-      end
-      builder.send('purchase-order-information') do
-        @element[:purchase_order_information].build(builder)
+    def build(builder)
+      builder.send('header') do
+        builder.send 'customer-id', IngramMicro.configuration.customer_id
+        builder.send 'business-name', @element[:business_name]
+        builder.send 'carrier-name', @element[:carrier_name]
+        builder.send('customer-information') do
+          @element[:customer].build(builder)
+        end
+        builder.send('shipment-information') do
+          @element[:shipment_information].build(builder)
+        end
+        builder.send('purchase-order-information') do
+          @element[:purchase_order_information].build(builder)
+        end
+        builder.send('credit-card-information') do
+          @element[:credit_card_information].build(builder)
+        end
+        builder.send('order-header') do
+          @element[:order_header].build(builder)
+        end
       end
-      builder.send('credit-card-information') do
-        @element[:credit_card_information].build(builder)
+      check_line_items
+      builder.send('detail') do
+        @element[:detail].build(builder)
       end
-      builder.send('order-header') do
-        @element[:order_header].build(builder)
-      end
-    end
-    check_line_items
-    builder.send('detail') do
-      @element[:detail].build(builder)
     end
-  end
 
-  def check_line_items
-    if @element[:detail].element[:line_items].empty?
-      line_item = IngramMicro::ReturnAuthorizationLineItem.new
-      @element[:detail].element[:line_items] << line_item
+    def check_line_items
```

Comment:
method does more than just `check`, should be renamed to convey that it does more